channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -239,18 +244,77 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControlsAI {
                         }
                     });
 
+                    // Only update the endpoint if the original config !== the current config
+                    // This is so any redirect endpointUrl is not overwritten
+                    if (_orgEndpointUrl !== senderConfig.endpointUrl) {
+                        if (_orgEndpointUrl) {
+                            // TODO: add doc to remind users to flush before changing endpoint
+                            _self._buffer.clear();
```

Comment:
I don't think we want to clear the buffer, as this could cause the loss of events, the TODO is valid but we just tell them that if they don't existing queued events will also be sent to the new endpoint.